### PR TITLE
Add installation and packaging targets [gilvv]

### PR DESCRIPTION
**Pull request by vaibhav17octo**
You just need to build the image and run the container:
docker build . -t cpack-exercise 
docker run -it --rm --name test-container cpack-exercise  

You will logged in to the terminal. The apt package is already installed. Everything is available in the build directory.

You can see the tree structure:
cd /mnt/cpack-exercise/cpackexample
tree

I was able to create the cmakeconfig file and also create and install the debian package.
image

I added the following line to make sure that stripping was being done while the creation of the package:
set(CPACK_STRIP_FILES TRUE)
Screenshot 2024-12-10 at 6 59 45 PM

When this was set to FALSE the resulting package's size was increased to 4,5MB
Screenshot 2024-12-10 at 7 01 57 PM

If you are not adding this in the cmakeconfig file, you can also set this at the time of cmake command like so:
cmake -DCMAKE_INSTALL_PREFIX:PATH=/mnt/cpack-exercise/cpackexample -D CPACK_STRIP_FILES=TRUE ..  

I had also tried doing the optional task which allows the host to have the tar.gz file and .deb package file on the host after the container exists.
I was able to figure out an alternative.

Run the docker container:
docker run -it --rm --name test-container cpack-exercise 

Open another terminal and run the following commands:
docker cp test-container-1:/mnt/cpack-exercise/build/cpackexample-0.1.0-Linux.deb .
docker cp test-container-1:/mnt/cpack-exercise/build/cpackexample-0.1.0-Linux.tar.gz . 

It doesn't really solve the automization problem but that is what I could do.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -25,4 +26,19 @@ ENV LIBRARY_PATH $LIBRARY_PATH:/usr/local/lib/
...
+
+#The commands build the code using cmake and make. It then installs and make the debian package 
+RUN mkdir build && cd build \
+    && cmake -DCMAKE_INSTALL_PREFIX:PATH=/mnt/cpack-exercise/cpackexample .. \ 
+    && make -j \
+    && make install \
+    && make package \
+    && apt install ./cpackexample-0.1.0-Linux.deb
+
+RUN useradd -ms /bin/bash myuser
```
Comment (uekerman) [inline Dockerfile:39]: Why is this necessary?

**Review by uekerman — COMMENTED** [review]
> cd /mnt/cpack-exercise/cpackexample

This folder does not exist for me.

> If you are not adding this in the cmakeconfig file, you can also set this at the time of cmake command like so:
cmake -DCMAKE_INSTALL_PREFIX:PATH=/mnt/cpack-exercise/cpackexample -D CPACK_STRIP_FILES=TRUE ..

This is true for any CMake variable. Such packaging instructions, however, are typically hard-coded, since the user does not need to interact with them.

**Comment by uekerman** [comment]
Running `cpackexample` within the container gives a deal.II error.

File: Dockerfile (hunk 1 above)
Comment (vaibhav17octo) [reply Dockerfile:39]: This is not necessary but running lintian with root gives you a warning. 
Running it with a user doesn't

**Comment by vaibhav17octo** [comment]
> Running `cpackexample` within the container gives a deal.II error.

It is because of the new user that I added. 
If you run it with root. The error isn't there.
